### DISCUSSION TITLE
Why is the UpdateStats "empty" after an update call?

### DISCUSSION THREAD

**Discussion by dragos** [sbt/sbt#7196] [Q&A]
I was trying to get some information about dependency resolution, and trying it out with the default Coursier resolver misses the UpdateStats record:

```
sbt:test-project> show update
..
[info] core / update
[info] 	Update report:
[info] 	Resolve time: -1 ms, Download time: -1 ms, Download size: -1 bytes
```

If I disable Coursier (`ThisBuild / useCoursier := false`), I get more meaningful stats:

```
[info] core / update
[info] 	Update report:
[info] 	Resolve time: 380 ms, Download time: 35 ms, Download size: 0 bytes
```

Is this a known gap in Coursier's functionality, or something that can be implemented on the Sbt side? I'd be happy to take a stab at it if I had a couple of pointers on where to look. I'm not very familiar with neither the Sbt nor Coursier code base.

**Comment by adpi2** [comment]
Yes I guess this is a gap in Coursier's functionality.

Here are a few pointers:
- the [Coursier implementation](https://github.com/coursier/sbt-coursier/blob/8a921c6872a61d062121df455ba0b9b97e996886/modules/lm-coursier/src/main/scala/lmcoursier/CoursierDependencyResolution.scala#L107-L112) of `update`
- It uses [here](https://github.com/coursier/sbt-coursier/blob/8a921c6872a61d062121df455ba0b9b97e996886/modules/lm-coursier/src/main/scala/lmcoursier/internal/ResolutionRun.scala#L82-L122) the Resolve API from Coursier, which returns a [Resolution](https://github.com/coursier/coursier/blob/33eac58f8a8695e47e4a1fe6d3a4c9951becc6b6/modules/core/shared/src/main/scala/coursier/core/Resolution.scala#L693-L710) object which, I guess, does not contain any stats.

To fix this I guess we should collect the stats during the resolution process and expose them to the user in the `Resolution` response. I encourage you to open an issue in Coursier to discuss this feature with its main maintainer.

I am sorry I cannot give you more than just suppositions, but I hope you will be able to find more precise answers exploring Coursier and playing with it a bit.

**Reply by dragos** [reply]
Thank you, these are good pointers.
